common.lic

```diff
@@ -471,6 +471,16 @@ module DRC
     end
   end
 
+  def check_listening(last_teacher)
+    return if DRRoom.pcs.empty?
+    classes = assess_teach
+    last_teacher = classes
```

Comment:
Is this assigning to the input var? I thought Ruby passed this by value, not ref. In either case, for clarity I would much prefer it to be returned instead.

Reply:
> I thought Ruby passed this by value, not ref.
It does, I think this is still residue from when it was copied from the original file.

I think this can become:
```ruby
# Find the first teacher who is teaching a class
# for a skill we need to prioritize learning then listen.
classes
  .reject { |teacher, skill| teacher.nil? || skill.nil? }
  .sort_by { |teacher, skill| [DRSkill.getxp(skill), DRSkill.getrank(skill)] }
  .find { |teacher, skill| listen?(teacher) }
```